src/ci/docker/scripts/musl-toolchain.sh

```diff
@@ -0,0 +1,64 @@
+set -ex
+
+hide_output() {
```

Comment:
I seem to recall this being in a `shared.sh` script somewhere intended for inclusion to avoid duplication, but I may be wrong and if this is duplicated elsewhere seems fine to leave here.

Reply:
Every scripts defines it because it's not defined in `shared.sh`. It'd be nice to deduplicate all the scripts but this is out of scope for this PR.

I'll address other comments.